### ISSUE TITLE
'Double URL Decode' option

### ISSUE THREAD

**Issue by stealzy** [einaregilsson/Redirector#87]
Playing with redirect, I found what some crazy people creating double encoded links.
Example link: https://yandex.ru/clck/redir/dtype=clck/lid=error404.desktop.sites.market.title/sid=1488115774.96005.22881.28902/rnd=1488116264328/*data=url%3Dhttps%253A%252F%252Fmarket.yandex.ru%252F%253Fclid%253D568
So, for redirect this links URL Decode not enough.

Offtop:
I love this extension, using it in combination with RedirectControl on Firefox.
Also, I still did not understand how to use wildcards, because there is not at least simple usage example, so I have to use RegEx and escaping every dot and question mark in URL so boring.

**Comment by einaregilsson** [comment]
Well, I think this is too specific to add it as a feature. I doubt you'll find more cases where links are double encoded, so I'm closing this one.

As for Wildcards, there's only one special character in them, the *, which is similar to .* in regex. E.g. https://*.cnn.com would match https://foo.cnn.com , https://bar.cnn.com etc.

**Comment by einaregilsson** [comment]
Well, I think this is too specific to add it as a feature. I doubt you'll find more cases where links are double encoded, so I'm closing this one.

As for Wildcards, there's only one special character in them, the *, which is similar to .* in regex. E.g. https://*.cnn.com would match https://foo.cnn.com , https://bar.cnn.com etc.

**Comment by Gitoffthelawn** [comment]
I think you meant "...the * which is similar to..." instead of "...the , which is similar to...".

**Comment by einaregilsson** [comment]
Haha, ok, the asterisk is significant in Markdown, I had to write it twice to make it show up.

**Comment by equokka** [comment]
Having this issue as well, trying to deal with [curseforge](https://www.curseforge.com/minecraft/) redirect links:
```
https://www.curseforge.com/linkout?remoteUrl=http%253a%252f%252fjourneymap.techbrew.net%252f
to
http://journeymap.techbrew.net/
```
![image](https://user-images.githubusercontent.com/51298563/68092778-42b4f280-fe7e-11e9-918d-fd8c7b428736.png)

I understand it's niche but hey. Here's another example of this behavior. Love the extension!

**Comment by einaregilsson** [comment]
Ok, I'll consider it for the next version :)

**Comment by Gitoffthelawn** [comment]
I think it's a good idea too.

BTW, if I remember correctly (and it's been several years, so I'm not completely sure), I seem to recall dealing with this successfully back before Redirector had mandatory recursion protection.

Basically, I had a rule that would detect encoded URL's and decode them.  Since recursion was allowed, any encoded output would get processed.  It was nice because the decoded output would then get processed by all the rules again, which would allow further processing.

So one solution *might* be to add a toggle to allow/disallow recursion.

**Comment by OhSoGood** [comment]
+1. I have the same situation on awin1.com, which is used by a lot of websites to track users. So double URL decoding would be very welcome! Thank you!!

**Comment by einaregilsson** [comment]
Ok, ok, I'll try to add it for the next version. No idea when that is though.

**Comment by equokka** [comment]
We can close this now ^^

**Comment by einaregilsson** [comment]
Fixed in v3.5.1.
